front/src/HOC/withLogger.test.js

```diff
@@ -0,0 +1,211 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withLogger } from './withLogger';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+const TestComponent = () => <></>;
+const TestClassComponent = () => {
+  const ComponentForTest = class MyClass extends Component {
+    componentDidMount() {
+      if (this.props.log) {
+        this.props.log('inner component mounted');
+      }
+    }
+    shouldComponentUpdate() {
+      if (this.props.log) {
+        this.props.log('inner component should update');
+      }
+      return true;
+    }
+    componentDidUpdate() {
+      if (this.props.log) {
+        this.props.log('inner component updated');
+      }
+    }
+    componentWillUnmount() {
+      if (this.props.log) {
+        this.props.log('inner component will unmount');
+      }
+    }
+    render() {
+      return <></>;
+    }
+  };
+  ComponentForTest.propsTypes = {
+    log: PropTypes.func,
+  };
+  ComponentForTest.defaultProps = {
+    log: () => {},
+  };
+  return ComponentForTest;
+};
+const fakeLogger = jest.fn(data => data);
+
+describe('Testing withLogger', () => {
+  afterEach(() => {
+    fakeLogger.mockClear();
```

Comment:
@superpike could you please explain why is it necessary, what was the problem and have you considered other solutions?

Reply:
As soon as we made fakeLogger function out of the each test, it's calls are accumulated and we got order dependency for tests. That's why we need to clear mock data